### PR TITLE
Detect inputs function

### PR DESCRIPTION
**Pull request by jamaliki**
Hello! This is my (attempted) solution for Issue #1 , please let me know if I should change or anything.

### CODE REVIEW

**Review by Makogan — CHANGES_REQUESTED** [review]
The code looks good if you make the last modification I pointed out.

File: Src/Engine/DataSanitation/AttributeInputSanitation.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "Shared/Shared.hpp"
+#include "Geometry/MeshConstraints.hpp"
+
+#include <concepts>
+
+template<typename InputType> uint32_t DetectInputs()
```
[the remark this answers is not in the document]
Comment (jamaliki) [reply Src/Engine/DataSanitation/AttributeInputSanitation.hpp]: I changed it to `Flags<AttributeTagBits>` which is what `AttributeTag` is defined as.

[the remark this answers is not in the document]
Comment (Makogan) [reply Src/Engine/DataSanitation/AttributeInputSanitation.hpp]: That is perfect, push the latest commit so that I can approve it.

**Review by Makogan — APPROVED** [review]
Perfect

**Pull request merged**